packages/x-bundle/src/cache/defs.ts

```diff
@@ -0,0 +1,104 @@
+import { Store } from "cache-manager";
+
+export type CacheStore = {/* prettier-ignore */} | Store | "memory";
+export type ICacheManagerConfig = {
+  store: CacheStore;
+  storeConfig: StoreConfig;
+  resolverDefaultConfig?: CacheOptions;
+};
+export type StoreConfig = (
+  | BasicStoreConfig
+  | MemcachedStoreConfig
+  | HazelcastStoreConfig
+  | RedisStoreConfig
+  | IoRedisStoreConfig
+  | MongoDbStoreConfig
+  | MongooseStoreConfig
+  | FsHashStoreConfig
+  | FsBinaryStoreConfig
+) & {
+  ttl?: number;
+  isCacheableValue?: (value: any) => boolean;
+  refreshThreshold?: number;
+};
+
+export type RedisStoreConfig = {
+  host: string;
+  port: number;
+  auth_pass: string;
+  db?: number;
+  url?: string;
+};
+export type IoRedisStoreConfig = {
+  host: string;
+  port: number;
+  password: string;
+  db?: number;
+  clusterConfig?: {
+    nodes: { port: number; host: string }[];
+    options: {
+      maxRedirections: number;
+    };
+  };
+};
+export type MongoDbStoreConfig = {
+  uri: string;
+  options?: {
+    collection?: string;
+    compression?: boolean;
+    poolSize?: number;
+    autoReconnect?: boolean;
+  };
+};
+export type MongooseStoreConfig = {

```

Comment:
do we still need these types, let's first offer defaeult for mongodb, memory, redis.

Reply:
if we delete them, we r gonna make the type of storeConfig any, is that ok ?